numpy/core/src/multiarray/datetime.c

```diff
@@ -973,6 +969,10 @@ convert_datetime_divisor_to_multiple(PyArray_DatetimeMetaData *meta,
         if (meta->base == NPY_FR_as) {
             num = 0;
         }
+    } else {
```

Comment:
```suggestion
    }
    else {
```
I have to admit, if you have a short comment for the `>= NPY_FR_s` branch, I would love that.  The fact that it modifies the global static was also confusing to me.

I wonder hard would it would be to set up one of the CI jobs to use an address sanitizer run?

Reply:
Fixed the formatting.

I added a comment, although I'm not sure it really adds a lot. I don't understand this code well enough to produce a better description off the top of my head: I was just sent here by an asan failure :-)

Yes, I would enthusiastically recommend setting up CI jobs that use sanitizers. We do that routinely and it saves a lot of time chasing bugs in code in memory-unsafe languages. I suspect it's easy enough to do: I believe a sufficiently modern `clang` or probably `gcc` has everything built in.